packages/block-library/src/cover/edit.js

```diff
@@ -316,6 +320,76 @@ function CoverEdit( {
 		allowedBlocks,
 		templateLock,
 	} = attributes;
+
+	const coverImage = useRef( {
+		url,
+		type: backgroundType,
+		isFeaturedImage: useFeaturedImage,
+	} );
+
+	const [ featuredImage ] = useEntityProp(
+		'postType',
+		postType,
+		'featured_media',
+		postId
+	);
+
+	const media = useSelect(
+		( select ) =>
+			featuredImage &&
+			select( coreStore ).getMedia( featuredImage, { context: 'view' } ),
+		[ featuredImage ]
+	);
+	const mediaUrl = media?.source_url;
+
+	useEffect( () => {
+		// Save the currently set image
+		if ( useFeaturedImage && url ) {
+			coverImage.current = {
+				url,
+				type: backgroundType,
+				isFeaturedImage: false,
+			};
+		}
+
+		// If the useFeaturedImage attribute is set
+		// but there is no featured media on the post
+		// we reset the media
+		if ( useFeaturedImage && ! mediaUrl ) {
+			setAttributes( {
+				url: undefined,
+				id: undefined,
+				backgroundType: undefined,
+				focalPoint: undefined,
+				hasParallax: undefined,
+				isRepeated: undefined,
+			} );
+		}
+
+		// The featured image is in use and it has changed
+		if ( mediaUrl && mediaUrl !== url && useFeaturedImage ) {
+			onSelectMedia( {
+				url: mediaUrl,
```

Comment:
Why are we keeping the url of the featured image in the attributes? shouldn't it be dynamic?

Reply:
That is a good point, mostly for convenience in the edit component as may behaviours depend on this attribute. We also need it when we auto-update the cover block if the post has a new featured image and the block is open again.